node/src/accountant/payment_adjuster/adjustment_runners.rs

```diff
@@ -0,0 +1,254 @@
+    use crate::accountant::test_utils::{
+        make_analyzed_account, make_non_guaranteed_qualified_payable,
+    };
+    use crate::accountant::{AnalyzedPayableAccount, CreditorThresholds, QualifiedPayableAccount};
+    use crate::sub_lib::wallet::Wallet;
+    use crate::test_utils::make_wallet;
+    use std::time::SystemTime;
+
+    fn initialize_payment_adjuster(
+        now: SystemTime,
+        service_fee_balance: u128,
+        largest_exceeding_balance_recently_qualified: u128,
+    ) -> PaymentAdjusterReal {
+        make_initialized_subject(
+            Some(now),
+            Some(service_fee_balance),
+            None,
+            Some(largest_exceeding_balance_recently_qualified),
+            None,
+        )
+    }
+
+    fn make_weighed_payable(n: u64, initial_balance_minor: u128) -> WeightedPayable {
+        let mut payable = WeightedPayable::new(make_analyzed_account(111), n as u128 * 1234);
+        payable
+            .analyzed_account
+            .qualified_as
+            .bare_account
+            .balance_wei = initial_balance_minor;
+        payable
+    }
+
+    fn test_surplus_incurred_after_disqualification_in_previous_iteration(
+        subject: ServiceFeeOnlyAdjustmentRunner,
+        payable_1: WeightedPayable,
+        payable_2: WeightedPayable,
+        cw_service_fee_balance_minor: u128,
+        expected_proposed_balance_1: u128,
+        expected_proposed_balance_2: u128,
+    ) {
+        // Explanation: The hypothesis is that the previous iteration disqualified an account after
+        // which the remaining means are enough for the other accounts.
+        // We could assign the accounts all they initially requested but a fairer way to do that
+        // is to give out only that much up to the disqualification limit of these accounts. Later
+        // on, the accounts that deserves it more, according to their ordering based on their former
+        // weights, will split the rest of the means among them (Their weights were higher).
+        let now = SystemTime::now();
+        let mut payment_adjuster =
+            initialize_payment_adjuster(now, cw_service_fee_balance_minor, 12345678);
+
+        let result = subject.adjust_accounts(
+            &mut payment_adjuster,
+            vec![payable_1.clone(), payable_2.clone()],
+        );
+
+        assert_eq!(
+            result,
+            vec![
+                AdjustedAccountBeforeFinalization::new(
+                    payable_1.analyzed_account.qualified_as.bare_account,
+                    payable_1.weight,
+                    expected_proposed_balance_1
+                ),
+                AdjustedAccountBeforeFinalization::new(
+                    payable_2.analyzed_account.qualified_as.bare_account,
+                    payable_2.weight,
+                    expected_proposed_balance_2
+                )
+            ]
+        )
+    }
+
+    fn weighted_payable_setup_for_surplus_test(
```

Comment:
You don't need this function at least inside the two tests. You can use the `make_weighted_payable` function and remove this function.